config/v2_3_experimental/types/disk.go

```diff
@@ -142,3 +148,17 @@ func (n Disk) partitionsMixZeroesAndNonexistence() bool {
 	}
 	return hasZero && hasShouldNotExist
 }
+
+func (n Disk) partitionsUnitsMismatch() bool {
```

Comment:
Could we add a `partionUnitsMismatch` that this calls for each partition so we could re-use the check in the `partition.Validate`?

Reply:
I don't think that would help. If units mismatch for a partition we want to tell them right there (higher level of granularity for the error message). But if they specify one partition entirely in megabytes and one entirely in sectors we still want to error on that.